cfme/control/explorer/alerts.py

```diff
@@ -304,6 +306,29 @@ def create(self, description, severity=None, active=None, based_on=None, evaluat
         view.flash.assert_success_message('Alert "{}" was added'.format(alert.description))
         return alert
 
+    def all(self):
+        # get alerts via reading 'All Alerts table'
+        view  = navigate_to(self.appliance.collections.alerts,'All')
+        try:
+            alert_table = view.table
+            alerts = [
+                self.instantiate(
+                      description=alert.get('Description'),
+                      active=alert.get('Active'),
+                      based_on=alert.get('Based On'),
+                      evaluate=alert.get('What is evaluated'),
+                      emails=alert.get('Email'),
+                      snmp_trap=alert.get('SNMP'),
+                      timeline_event=alert.get('Event on Timeline'),
+                      mgmt_event=alert.get('Management Event Raised')
+                )
+                for alert in alert_table.read()
+            ]
+        except AttributeError:
```

Comment:
So just did some testing here, I believe this is not an `AttributeError` but a `NoSuchElementException`

Reply:
Hmm... what steps did you take to get that Exception? If I do 
```
view = navigate_to(appliance.collections.alert_profiles,'All')
view.table.read()
``` 

I get `AttributeError: 'AlertProfilesAllView' object has no attribute 'table' `

Reply:
That is because the `AlertsProfileAllView` does not have a table attribute. This is what I did:
```
view  = navigate_to(appliance.collections.alerts,'All')
view.table.read()
<ignoring the output here since it is a dump of the table>

Now I navigated to another page that doesn't have that defined table (to simulate the page rendering but with no table)
ocp = providers.get_crud('ocp-39-hawk')
In[5]: navigate_to(ocp,'All')
Out[5]: <cfme.common.provider_views.ContainerProvidersView at 0x7fc0a555bd50>
In[6]: view.table.read()
Traceback (most recent call last):
  File "/home/juwatts/repos/cfme_venv/lib/python2.7/site-packages/IPython/core/interactiveshell.py", line 2878, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-6-413af58f8186>", line 1, in <module>
    view.table.read()
  File "/home/juwatts/repos/cfme_venv/lib/python2.7/site-packages/widgetastic/log.py", line 117, in wrapped
    result = f(self, *args, **kwargs)
  File "/home/juwatts/repos/cfme_venv/lib/python2.7/site-packages/widgetastic/widget/table.py", line 782, in read
    rows = list(self)
  File "/home/juwatts/repos/cfme_venv/lib/python2.7/site-packages/widgetastic/widget/table.py", line 525, in _all_rows
    for row_pos in range(1, self.row_count + 1):
  File "/home/juwatts/repos/cfme_venv/lib/python2.7/site-packages/widgetastic/widget/table.py", line 856, in row_count
    return len(self.browser.elements(self.ROWS, parent=self))
  File "/home/juwatts/repos/cfme_venv/lib/python2.7/site-packages/widgetastic/browser.py", line 826, in elements
    force_check_safe=force_check_safe)
  File "/home/juwatts/repos/cfme_venv/lib/python2.7/site-packages/widgetastic/utils.py", line 682, in wrap
    return method(*args, **kwargs)
  File "/home/juwatts/repos/cfme_venv/lib/python2.7/site-packages/widgetastic/browser.py", line 260, in elements
    root_element = self.element(parent, parent=parent.locatable_parent)
  File "/home/juwatts/repos/cfme_venv/lib/python2.7/site-packages/widgetastic/browser.py", line 327, in element
    elements = self.elements(locator, *args, **kwargs)
  File "/home/juwatts/repos/cfme_venv/lib/python2.7/site-packages/widgetastic/utils.py", line 682, in wrap
    return method(*args, **kwargs)
  File "/home/juwatts/repos/cfme_venv/lib/python2.7/site-packages/widgetastic/browser.py", line 249, in elements
    locator = self._process_locator(locator)
  File "/home/juwatts/repos/cfme_venv/lib/python2.7/site-packages/widgetastic/browser.py", line 198, in _process_locator
    return locator.__element__()
  File "/home/juwatts/repos/cfme_venv/lib/python2.7/site-packages/widgetastic/widget/base.py", line 66, in wrapped
    return method(self, *new_args, **new_kwargs)
  File "/home/juwatts/repos/cfme_venv/lib/python2.7/site-packages/widgetastic/widget/base.py", line 341, in __element__
    return self.parent_browser.element(locator)
  File "/home/juwatts/repos/cfme_venv/lib/python2.7/site-packages/widgetastic/browser.py", line 337, in element
    raise NoSuchElementException('Could not find an element {}'.format(repr(locator)))
NoSuchElementException: Message: Could not find an element Locator(by='xpath', locator=u".//div[@id='alert_list_div' or @class='miq-data-table']/table")

```